kazoo/protocol/connection.py

```diff
@@ -1,12 +1,14 @@
 """Zookeeper Protocol Connection Handler"""
 from binascii import hexlify
 from contextlib import contextmanager
+
 import logging
 import random
 import select
 import socket
 import sys
 import time
+from puresasl.client import SASLClient
```

Comment:
Since this is a 3rd-party lib, it should be an optional dependency (which you correctly handled in the `setup.py`)... So shouldn't this be wrapped in a `try`/`except` to handle the `ImportError` when this library is not available?